hudi-client/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -116,6 +119,19 @@ HoodieCleanerPlan requestClean(JavaSparkContext jsc) {
         PartitionCleanStat partitionCleanStat = partitionCleanStatMap.get(partitionPath);
         partitionCleanStat.addDeleteFilePatterns(deletePath.getName());
         partitionCleanStat.addDeletedFileResult(deletePath.getName(), deletedFileResult);
+
+        // If CleanBootstrapSourceFileEnabled and it is a metadata bootstrap commit, also delete the corresponding source file
+        if (cleanBootstrapSourceFileEnabled && !FSUtils.isLogFile(deletePath)
+            && FSUtils.getCommitTime(delFileName).equals(HoodieTimeline.METADATA_BOOTSTRAP_INSTANT_TS)) {
+          Option<HoodieBaseFile> baseFile = fileSystemView.getBaseFileOn(partitionPath,
```

Comment:
@zhedoubushishi : I think this is an unnecessary call we will be making per file deletion. With embedded timeline service, this request will be routed to driver. Instead, I think we can handle it using versioning in clean plan.  Thoughts ?

Reply:
I see what you mean. Yea we should avoid making unnecessary calls. So regarding to versioning, do you mean create a class like [TimelineLayoutVersion](https://github.com/apache/hudi/blob/5ee676e34fa2429d02ecd1a3cf7f586518ed2081/hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/TimelineLayoutVersion.java#L29) or just simply handle this logic in the ```CleanActionExecutor```. For example, if the parameter is a full file path then delete it. Else if the parameter is a file name, generate the file path and then delete it.

Reply:
@zhedoubushishi : Tried reaching you on slack :) to discuss an approach. I went ahead and implemented it in the interest of time. 

The basic idea is to ensure that Cleaner plan stores necessary information related to files to be deleted including bootstrap base files. The cleaner executor will simply read the cleaner plan and be able to distinguish normal vs bootstrap base files. It goes ahead and deletes those files. For bootstrap base files, it records the complete path of the file it deleted in a separate (new) avro field. This is very important in order to ensure incremental timeline syncing (which reads these metadata) to work properly. 

Please take a look at this code changes if possible. 
( @vinothchandar  @umehrot2  : FYI )

Reply:
@zhedoubushishi : Please also note that for ensuring backwards compatibility, I add a migrator class (CleanPlanMigrator.java) to handle pre and post 0.6  .clean.requested files.